pkg/ingester/client/circuitbreaker.go

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package client
+
+import (
+	"context"
+	"errors"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/sony/gobreaker"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/grafana/mimir/pkg/util/spanlogger"
+)
+
+const (
+	resultSuccess = "success"
+	resultError   = "error"
+	resultOpen    = "circuit_breaker_open"
+)
+
+func NewCircuitBreaker(name string, cfg CircuitBreakerConfig, metrics *Metrics, logger log.Logger) grpc.UnaryClientInterceptor {
```

Comment:
` name` is actually the ingester address. What do you think if we rename it for clarity? I would also rename it in the logs from `"name", name` to `"addr", addr`.

Reply:
Fixed in e9fc92026c936af1eb2d9d83aad333f1a31ddbda